flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/decorators/ServiceDecorator.java

```diff
@@ -89,7 +90,13 @@ protected Service decorateInternalResource(Service resource, Configuration flink
 		}
 
 		spec.setPorts(servicePorts);
-		spec.setSelector(resource.getMetadata().getLabels());
+
+		final Map<String, String> labels = new LabelBuilder()
```

Comment:
`decorateInternalResource` is only used for job manager?
I think we should add a unit test for covering the changes if it is really a bug. I mean before the changes the test should fail, and after this change it can success. Otherwise we can not measure what is the difference.

Reply:
I have update the test `testCreateInternalService` and `testCreateRestService` to verify the expected selector labels. Jobmanager component label should exist.

If you revert the change of `ServiceDecorator`, the updated tests will fail.